flake.nix

```diff
@@ -3,88 +3,153 @@
 
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    treefmt-nix = {
+      url = "github:numtide/treefmt-nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
```

Comment:
In HM I got backlash introducing the dependency to the flake inputs. Opted for a more manual approach to prevent users from having to grab dev dependencies when they are just consuming the flake and not developing for it. 

```nix
        formatter = forAllPkgs (
          pkgs:
          pkgs.treefmt.withConfig {
            runtimeInputs = with pkgs; [
              nixfmt-rfc-style
              deadnix
              keep-sorted
            ];
            settings = pkgs.lib.importTOML ./treefmt.toml;
          }
        );
```